website/static/js/accountSettings.js

```diff
@@ -152,42 +154,54 @@ var UserProfileViewModel = oop.defclass({
         );
     },
     addEmail: function () {
-        var email = new UserEmail({
-            address: this.emailInput().toLowerCase().trim()
-        });
+        this.message('');
+        this.messageClass('text-info');
+        var newEmail = this.emailInput().toLowerCase().trim();
+        if(newEmail){
+            var email = new UserEmail({
+                address: newEmail
+            });
 
-        // ensure email isn't already in the list
-        for (var i=0; i<this.profile().emails().length; i++) {
-            if (this.profile().emails()[i].address() === email.address()) {
-                $osf.growl('Error', 'Duplicate Email', 'warning');
-                this.emailInput('');
-                return;
+            // ensure email isn't already in the list
+            for (var i=0; i<this.profile().emails().length; i++) {
+                if (this.profile().emails()[i].address() === email.address()) {
+                    $osf.growl('Error', 'Duplicate Email', 'warning');
+                    this.emailInput('');
+                    return;
+                }
             }
-        }
 
-        this.profile().emails.push(email);
+            this.profile().emails.push(email);
 
-        this.client.update(this.profile()).done(function (profile) {
-            this.profile(profile);
+            this.client.update(this.profile()).done(function (profile) {
+                this.profile(profile);
 
-            var emails = profile.emails();
-            for (var i=0; i<emails.length; i++) {
-                if (emails[i].address() === email.address()) {
-                    this.emailInput('');
-                    $osf.growl('<em>' + email.address()  + '<em> added to your account.','You will receive a confirmation email at <em>' + email.address()  + '<em>. Please check your email and confirm.', 'success');
-                    return;
+                var emails = profile.emails();
+                for (var i=0; i<emails.length; i++) {
+                    if (emails[i].address() === email.address()) {
+                        this.emailInput('');
+                        $osf.growl('<em>' + email.address()  + '<em> added to your account.','You will receive a confirmation email at <em>' + email.address()  + '<em>. Please check your email and confirm.', 'success');
+                        return;
+                    }
                 }
-            }
-            $osf.growl('Error', 'Email validation failed', 'danger');
-        }.bind(this));
+                $osf.growl('Error', 'Email validation failed', 'danger');
+            }.bind(this));
+        } else {
+            this.message('Email cannot be empty.');
+            this.messageClass('text-danger');
```

Comment:
address in  e20b99d bde4ff9
Will need to refactor the other part codes to use the ChangeMessage.js